### PR TITLE
Rollout video of variable dynamics

### PR DESCRIPTION
**Pull request by sahahner**
Create a video of a forecast of several variables starting from a validation sample.

This functionality builds up on the LongRolloutPlot function, which saves snapshots after longer rollouts. 

The number of rollout steps can be higher than the number of rollout steps used for training.
To allow longer rollouts in validation, the validation batches are made longer. This has no effect on the actual batch size.
The runtime to do the rollout is low, but creating multiple scatter plots and saving them to a gif takes around a minute per variable for a 14-day rollout of 6-hourly forecasts. 
Therefore, by **default**, the video rollout is **not enabled**.
If enabled, the default is to do a 56-step forecast video every 20 epochs.

Possible reviewers: @lzampier, @mc4117, @HCookie

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ Keep it human-readable, your future self will thank you!
 - Fix: Inference checkpoints are now saved according the frequency settings defined in the config [#37](https://github.com/ecmwf/anemoi-training/pull/37)
 - Feature: Add configurable models [#50](https://github.com/ecmwf/anemoi-training/pulls/50)
 - Feature: Support training for datasets with missing time steps [#48](https://github.com/ecmwf/anemoi-training/pulls/48)
+- Long rollout plots and videos (GIFS)
```
Comment (mchantry) [inline CHANGELOG.md:28]: Please can you add #65 to the line

File: src/anemoi/training/data/datamodule.py [hunk 2]
```diff
@@ -125,8 +125,12 @@ def ds_valid(self) -> NativeGridDataset:
         r = self.rollout
         if self.config.diagnostics.eval.enabled:
             r = max(r, self.config.diagnostics.eval.rollout)
-        if self.config.diagnostics.plot.get("longrollout") and self.config.diagnostics.plot.longrollout.enabled:
+        if self.config.diagnostics.plot.get("longrollout") and self.config.diagnostics.plot.longrollout.get("enabled"):
```
Comment (mchantry) [inline src/anemoi/training/data/datamodule.py:128]: Where is longrollout defined?

**Comment by mchantry** [comment]
Neat PR, what's the cost overhead of this?

**Comment by HCookie** [comment]
Hopefully it won't be much work, but the callbacks are being refactored in parallel,
https://github.com/ecmwf/anemoi-training/pull/60
Likely that will be merged in before this. So it should be as simple as pulling this callback into the appropriate sub module

**Comment by sahahner** [comment]
> Neat PR, what's the cost overhead of this?

Regarding the cost, I added an estimation for the default setup to the description of this PR.

**Comment by sahahner** [comment]
> Hopefully it won't be much work, but the callbacks are being refactored in parallel, #60 Likely that will be merged in before this. So it should be as simple as pulling this callback into the appropriate sub module

Thank you for the heads-up. I can merge after #60.

**Comment by HCookie** [comment]
@sahahner If this PR is ready before #60, please feel free to change the target branch of this to merge into #60.

**Comment by HCookie** [comment]
Do you have an example of the gif as saved in mlflow?

**Comment by sahahner** [comment]
> Do you have an example of the gif as saved in mlflow?

Have a look at this one: https://mlflow.ecmwf.int/#/experiments/23/runs/6172b5513408422f967895ecbe6101c2/artifacts

File: src/anemoi/training/config/diagnostics/plot/rollout_eval.yaml [hunk 3]
```diff
@@ -63,6 +63,7 @@ callbacks:
   - _target_:  anemoi.training.diagnostics.callbacks.plot.LongRolloutPlots
     rollout:
       - ${dataloader.validation_rollout}
+    video_rollout: ${dataloader.validation_rollout}
```
Comment (mc4117) [inline src/anemoi/training/config/diagnostics/plot/rollout_eval.yaml:66]: Do we not potentially want the video to rollout to longer than the normal rollout? If I understand correctly you can choose how frequently this is run. I'm imagining that every 3/4 epochs I might want a gif out to 10 days but with the other validation plots maybe I would only want out to 3 days?

Comment (sahahner) [reply src/anemoi/training/config/diagnostics/plot/rollout_eval.yaml:66]: Yes, you are right. We allow the evaluation on longer rollouts than training rollout.`dataloader.validation_rollout` is an additional variable that has been introduced in the [callback refactor](https://github.com/ecmwf/anemoi-training/pull/60) that enables evaluation on up to `dataloader.validation_rollout` timesteps.

Comment (mc4117) [reply src/anemoi/training/config/diagnostics/plot/rollout_eval.yaml:66]: what I meant was a `validation_rollout` and a `validation_video_rollout` which was longer. But maybe this isn't something desirable

Comment (sahahner) [reply src/anemoi/training/config/diagnostics/plot/rollout_eval.yaml:66]: You can set the rollout for validation metrics, plots and the validation video to any number below `dataloader.validation_rollout`. The configuration `dataloader.validation_rollout` makes sure that the validation batches are long enough to perform the validations over longer rollout.

Comment (mc4117) [reply src/anemoi/training/config/diagnostics/plot/rollout_eval.yaml:66]: Discussed and clarified offline

**Review by lzampier — APPROVED** [review]

**Pull request merged**